tests_repo_state/test_makefile.py

```diff
@@ -0,0 +1,49 @@
+from __future__ import unicode_literals
+
+import os
+import re
+import codecs
+
+import pytest
+
+def get_file_content(path):
+    """
+    Get the content of the UTF-8 text file at the specified path.
+    Used for pytest fixtures.
+    """
+    if not os.path.exists(path):
+        return ''
+    with codecs.open(path, 'r', 'utf-8') as open_file:
+        return open_file.read()
+
+@pytest.fixture
+def get_makefile(repo_path):
+    """Fixture containing the text content of Makefile"""
+    #TODO(jinder): make below work with inputs with both "/" at end and not
+    full_path = repo_path + '/Makefile'
+    return get_file_content(full_path)
+
+@pytest.fixture(scope="module")
+def module_data_holder(data_holder):
+    if 'makefile' not in data_holder:
+        data_holder['makefile'] = {}
+    return data_holder
+
+
+def test_makefile_exists(get_makefile, module_data_holder):
+    """
+    Test to check if repo hase Makefile
+    """
+    module_data_holder['makefile']['exists'] = False
+    if len(get_makefile) > 0:
+        module_data_holder['makefile']['exists'] = True
+
+def test_has_upgrade(get_makefile, module_data_holder):
+    """
+    Test to check if makefile has an upgrade target
+    """
+    regex_pattern = "upgrade:"
```

Comment:
I keep hoping I'll find a well-maintained Makefile parser in python, but no such luck yet.  The best I've found so far is [pymake](https://github.com/linuxlizard/pymake), which I suspect will fail badly on real-world Makefiles and hasn't been updated in 3 years.